subsys/bluetooth/shell/audio.c

```diff
@@ -1585,3 +1585,86 @@ static int cmd_audio(const struct shell *sh, size_t argc, char **argv)
 
 SHELL_CMD_ARG_REGISTER(audio, &audio_cmds, "Bluetooth audio shell commands",
 		       cmd_audio, 1, 1);
+
+ssize_t audio_ad_data_add(struct bt_data *data_array, const size_t data_array_size,
+			  const bool discoverable, const bool connectable)
+{
+	static const uint8_t ad_ext_uuid16[] = {
+		IF_ENABLED(CONFIG_BT_MICP_MIC_DEV, (BT_UUID_16_ENCODE(BT_UUID_MICS_VAL),))
+		IF_ENABLED(CONFIG_BT_ASCS, (BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL),))
+		IF_ENABLED(CONFIG_BT_BASS, (BT_UUID_16_ENCODE(BT_UUID_BASS_VAL),))
+		IF_ENABLED(CONFIG_BT_PACS, (BT_UUID_16_ENCODE(BT_UUID_PACS_VAL),))
+		IF_ENABLED(CONFIG_BT_GTBS, (BT_UUID_16_ENCODE(BT_UUID_GTBS_VAL),))
+		IF_ENABLED(CONFIG_BT_TBS, (BT_UUID_16_ENCODE(BT_UUID_TBS_VAL),))
+		IF_ENABLED(CONFIG_BT_VCS, (BT_UUID_16_ENCODE(BT_UUID_VCS_VAL),))
+		IF_ENABLED(CONFIG_BT_HAS, (BT_UUID_16_ENCODE(BT_UUID_HAS_VAL),))
+	};
+	size_t ad_len = 0;
+
+	if (!discoverable) {
+		return 0;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_ASCS)) {
+		static uint8_t ad_bap_announcement[8] = {
+			BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL),
+			BT_AUDIO_UNICAST_ANNOUNCEMENT_TARGETED,
+		};
+		enum bt_audio_context snk_context, src_context;
+
+		snk_context = bt_audio_capability_get_available_contexts(BT_AUDIO_DIR_SINK);
+		ad_bap_announcement[3] = (((snk_context) >> 0) & 0xFF);
+		ad_bap_announcement[4] = (((snk_context) >> 8) & 0xFF);
```

Comment:
Consider using something like `sys_put_le16` to insert values in arrays

Reply:
Ah, yes... Too much copy-paste :D Thanks!